lib/tokenize.cpp

```diff
@@ -5685,14 +5685,28 @@ void Tokenizer::simplifyPlatformTypes()
 
     /** @todo This assumes a flat address space. Not true for segmented address space (FAR *). */
 
-    if (_settings->sizeof_size_t == _settings->sizeof_long)
-        type = isLong;
-    else if (_settings->sizeof_size_t == _settings->sizeof_long_long)
-        type = isLongLong;
-    else if (_settings->sizeof_size_t == _settings->sizeof_int)
-        type = isInt;
+    if (_settings->platformType == Settings::Win32A || _settings->platformType == Settings::Win32W)
+    {
+        if (_settings->sizeof_size_t == _settings->sizeof_int)
```

Comment:
hmm I am not sure why we have this condition. If the platform is win32A or win32W then we know that sizeof(size_t) is equal to sizeof(int) right?

Reply:
No. It's true if the size_t and int are equal.
for example
i declare a integer int a = 8;
and int b = 6;
then sizeof (a) will be different from sizeof (b)

Reply:
I do not understand your answer. the value of a variable is not used when sizeof is calculated.